webapp/src/components/systemConsole/index.tsx

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useMemo, useState} from 'react';
+
+import './style.scss';
+import EnableSurvey from 'components/systemConsole/enableSurvey';
+import Expiry from 'components/systemConsole/expiry/expiry';
+import Questions from 'components/systemConsole/questions/questions';
+import SurveyDateTime from 'components/systemConsole/surveyDateTime/surveyDateTime';
+import TeamFilter from 'components/systemConsole/teamFilter/teamFilter';
+
+import type {CombinedConfig, CustomComponentProps, CustomConfigTypes} from 'types/mattermost-webapp';
+
+export type CustomSettingChildComponentProp = CustomComponentProps & {
+    setInitialSetting: (settingId: string, settings: CustomConfigTypes) => void;
+}
+
+function SystemConsoleSetting(props: CustomComponentProps) {
+    const {id, onChange, setSaveNeeded} = props;
+
+    // This holds the combined config of all sub-configs
+    const [config, setConfig] = useState<CombinedConfig>({} as CombinedConfig);
+
+    const onChangeWrapper = useCallback((settingId: string, settings: CustomConfigTypes) => {
+        const newConfig = {
+            ...config,
+            [settingId]: settings,
+        };
+
+        setConfig(newConfig);
+        onChange(id, newConfig);
+        setSaveNeeded();
+    }, [config, id, onChange, setSaveNeeded]);
+
+    // This is called by sub-config components on mount.
+    // Every sub-config component sets its config either from the saved values
+    // or the default values. Whatever they do, they are expected
+    // to call this with the settings they decide are to be set initially.
+    //
+    // This is needed so that on change in one sub-setting, we know all other sub-settings as well and thus
+    // can save the entire setting object on server side.
+    // Otherwise the displayed default values will never get saved unless a user changes them.
+    const setDefaults = useCallback((settingId: string, settings: CustomConfigTypes) => {
+        setConfig((existingConfig) => ({
+            ...existingConfig,
+            [settingId]: settings,
+        }));
+    }, []);
+
+    const modifiedProps = useMemo((): CustomSettingChildComponentProp => ({
+        ...props,
+        onChange: onChangeWrapper,
+        setInitialSetting: setDefaults,
+    }), [onChangeWrapper, props, setDefaults]);
```

Comment:
You are expanding the object, so in terms of memoization this is not needed. If you did it in terms of removing stress for the garbage collector, I could think it deeper whether it makes sense or not.